src/deps/binary/base64-browser.js

```diff
@@ -3,6 +3,9 @@ var thisAtob = function (str) {
 };
 
 var thisBtoa = function (str) {
+  if (/[\u0080-\uFFFF]/.test(str)) {
+    str = unescape(encodeURIComponent(str));
+  }
```

Comment:
> is the if statement necessary?

imho, it will speed up code execution for a simple string

> Actually now that I think about it, is it more reasonable to ask users to do this themselves?

non-ascii characters can get on btoa() not only on authorization. [The "Unicode Problem"](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#Solution_.232_.E2.80.93_rewriting_atob()_and_btoa()_using_TypedArrays_and_UTF-8)

>  if the user provides a string which is already unescape(encodeURIComponent())

double escaping will cause problems, but the rejection of the escaping throws DOMError